### PR TITLE
refactor: make more params configurable for smtp connector

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Make more parameters configurable for the SMTP connector.

## Testing

Tested locally.

### CODE REVIEW

File: packages/connector-smtp/src/index.ts [hunk 1]
```diff
@@ -32,22 +33,15 @@ const sendMessage =
...
-      // Enable `logger` to help debugging.
-      logger: true,
-    };
+    const configOptions: SMTPTransport.Options = config;
 
     const transporter = nodemailer.createTransport(configOptions);
 
+    // See https://nodemailer.com/smtp/proxies/#2-using-socks-proxy.
+    if (config.proxy?.startsWith('socks')) {
+      transporter.set('proxy_socks_module', socks);
```
Comment (wangsijie) [inline packages/connector-smtp/src/index.ts:42]: Wondering is it properly to add this proxy configs?

Comment (darcyYe) [reply packages/connector-smtp/src/index.ts:42]: I followed the guides of nodemailer and tested locally, it worked fine.
image

Comment (wangsijie) [reply packages/connector-smtp/src/index.ts:42]: My point is that do we need to implement the "proxy" part.

Comment (wangsijie) [reply packages/connector-smtp/src/index.ts:42]: @logto-io/eng

**Review by wangsijie — APPROVED** [review]

**Pull request merged**